tests/MyGame/Example/AbilityT.java

```diff
@@ -0,0 +1,27 @@
+// automatically generated by the FlatBuffers compiler, do not modify
+
+package MyGame.Example;
+
+import java.nio.*;
+import java.lang.*;
+import java.util.*;
+import com.google.flatbuffers.*;
+
+public class AbilityT {
+  private long id;
+  private long distance;
+
+  public long getId() { return this.id; }
```

Comment:
Lets not have getters and setters, but instead just make fields public. Remember these classes are pure "data", there are no methods that need to enforce invariants that would require private fields. Making them public also makes their use simpler.

Reply:
OK. Agreed.
Will change that.

Reply:
>Lets not have getters and setters, but instead just make fields public. Remember these classes are pure "data", there are no methods that need to enforce invariants that would require private fields. Making them public also makes their use simpler.

Although I fully agree with the rationale when analysing it in isolation, I think the "Java standards" of having getters/setters should be taking into consideration as well. This is the API that developers expects and we already have an initial friction for have an API being "a little different". There is no performance penalty and would integrates well with other languages on top of java, like Kotlin.

Also, the regular data classes generated have getters and setters, so I would assume as an user of the API, that the object-API version of it should like very similar.

Reply:
I think that we can go either way if we consider that anyway this object will be just for data storage, and if someone wants to enforce more restrictions on it he can just wrap it within another object.

Reply:
@aardappel @paulovap I need a decision regarding that.
Another option I thought is to add a cmd flag to add getters/setters to ObjectAPI classes. Yet, I need some agreement about that. Don't want to do additional job which might go to waste.

Reply:
@paulovap do people always use getters/setters in Java for "pure data" structs that will never have any other methods?

Reply:
>@paulovap do people always use getters/setters in Java for "pure data" structs that will never have any other methods?

Saying it is always like this is a strong statement, but vast majority APIs I used in all those years uses getters/setters for data classes. Personally I think it is a cargo-cult thing that came with JavaBean, and I challenged it in the past but that only caused pain.

The fact is that this is what users expect to see on their IDE's autocomplete and otherwise usually causes friction, which is something that our API already have too much (steep learning curve).

That being said, lot of tooling expect getters, When using Java from Kotlin, kotlin expects getters and setter with the same name and transforms it into a property (e.g `X.setId()`, `X.getId()` on a java class becomes `X.id` on Kotlin.

The new [record](https://openjdk.java.net/jeps/359) type on Java 14 generates getters automatically (although it is for immutable types, you might argue).

So given that those getters are inlined anyway, and this is what the users/tooling expects by default, I don't see a strong reason to get rid of it.

@aardappel decisions is yours of course, but wanted to give my point of view.

If you want to dive in the swamp(I wouldn't) you can check this [reddit thread](https://www.reddit.com/r/java/comments/2pqq1r/why_are_public_fields_so_demonized_in_java/) on why public fields are demonized in java.

Reply:
I am definitely not up to date with what is normal in the Java eco-system, so @paulovap 's opinion weighs heavier here :)
Lets keep the getters/setters then.

Reply:
Great. Marking as resolved.